### DISCUSSION TITLE
Capturing Code Executor Output as JSON for UI Integration in AutoGen

### DISCUSSION THREAD

**Discussion by mohan-enmovil** [microsoft/autogen#4313] [Q&A]
I have a use case where a user asks a question, and a code_writer agent generates a Python script with a MongoDB query. I want to use a code executor to run the script, capture the query output as JSON, and send it to a UI application.

From the examples I’ve seen, code executors often print the output. Is it possible to capture the output directly in JSON format and pass it to an external application like a UI? If so, please point me to example.

**Comment by ekzhu** [comment]
We don't have an example for this, but you can take a look at the code executors (not agents) in autogen-ext package. The code executors return structured types that can be serialized to JSON.
